src/controllers/author.controller.ts

```diff
@@ -26,25 +54,107 @@ export const authorDetail = asyncHandler(async (req: Request, res: Response, nex
 });
 
 export const authorCreateGet = asyncHandler(async (req: Request, res: Response, next: NextFunction) => {
-    res.send('[NOT IMPLEMENTED] Author is created with GET method.');
+    res.render('authors/form', { title: req.t('author.title.create') });
 });
 
-export const authorCreatePost = asyncHandler(async (req: Request, res: Response, next: NextFunction) => {
-    res.send('[NOT IMPLEMENTED] Author is created with POST method.');
-});
+export const authorCreatePost = [
+    ...validateAuthorForm,
+    asyncHandler(async (req: Request, res: Response, next: NextFunction) => {
+        const errors = validationResult(req);
+        const { first_name, family_name, date_of_birth, date_of_death } = req.body;
+        if (!errors.isEmpty()) {
+            res.render("authors/form", {
+                title: req.t("author.title.create"),
+                author: req.body,
+                errors: errors.array(),
+            });
+        }
+        else {
+            const author = await authorDAO.createAuthor(
+                first_name,
+                family_name,
+                date_of_birth,
+                date_of_death
+            );
+            res.redirect(`/authors/${author.id}`);
+        }
+    }), 
+];
 
 export const authorDeleteGet = asyncHandler(async (req: Request, res: Response, next: NextFunction) => {
-    res.send(`[NOT IMPLEMENTED] Author ${req.params.id} is deleted with GET method.`);
+    const id = parseInt (req.params.id);
+    if (isNaN(id)) {
+        return res.render('error', { message: req.t('error.Invalid') });
+    }
+    const author = await authorDAO.getAuthorById(id);
+    if (!author) {
+        return res.render('error', { message: req.t('error.NotFound') });
+    }
```

Comment:
move phần xử lý sang function not export để dùng chung cho các action detail, update, ... nhé